test/chainrules.jl

```diff
@@ -2,6 +2,7 @@
     @testset "BlockDiagonal" begin
         x = [randn(1, 2), randn(2, 2)]
         test_rrule(BlockDiagonal, x)
+        test_rrule(BlockDiagonal, x; output_tangent=Tangent{BlockDiagonal}(;blocks=x))
```

Comment:
Why does this work? I would have thought that the `blocks` field of the `output_tangent` would need to be `x.blocks` rather than `x`.

Reply:
x are the blocks :)

Reply:
🤦 right you are!